common/config/config.go

```diff
@@ -576,13 +591,51 @@ type (
 		// URI is the domain default URI for visibility archiver
 		URI string `yaml:"URI"`
 	}
+
+	// YamlNode is a lazy-unmarshaler, because *yaml.Node only exists in gopkg.in/yaml.v3, not v2,
+	// and go.uber.org/config currently uses only v2.
+	YamlNode struct {
+		unmarshal func(out any) error
+	}
 )
 
 const (
 	// NonShardedStoreName is the shard name used for singular (non-sharded) stores
 	NonShardedStoreName = "NonShardedStore"
+
+	FilestoreConfig = "filestore"
+	S3storeConfig   = "s3store"
+	GCloudConfig    = "gstorage"
 )
 
+var _ yaml.Unmarshaler = (*YamlNode)(nil)
```

Comment:
can you add unit test for new functions here?

Reply:
currently being tested via the config `Load`-using tests, which I think does cover this in this package... but lemme check